### DISCUSSION TITLE
New project fails with errors

### DISCUSSION THREAD

**Discussion by irdublu** [PiranhaCMS/piranha.core#1494] [Q&A]
I installed the Razor template and it runs fine (to the best of my limited knowledge).  I have created my own project from scratch to mirror the template and so get a better understanding.  Both are in the one solution so I switch the 'Set as Startup Project'.  I closed VS, deleted the bin and obj folders and now always get the errors shown in the Errors.jpg.  This is new.   I used to get Error 404 or the browser returned nothing.  Even _ViewImports has an error now yet my code and packages are identical (apart from Namespaces) to the template.  Can anyone help please?

![_ViewImports with error](https://user-images.githubusercontent.com/19719089/107078023-9f236200-67e5-11eb-837b-72ae63b2911c.JPG)

![Errors](https://user-images.githubusercontent.com/19719089/107078025-9fbbf880-67e5-11eb-9658-f6b3578284ca.jpg)

![Page cs](https://user-images.githubusercontent.com/19719089/107078026-9fbbf880-67e5-11eb-92db-8a70c5d76625.jpg)

![StandardPage cs](https://user-images.githubusercontent.com/19719089/107078027-9fbbf880-67e5-11eb-92af-c5a5209c54bc.jpg)

**Comment by GangsterGeek89** [comment]
![](https://user-images.githubusercontent.com/19719089/107078027-9fbbf880-67e5-11eb-92af-c5a5209c54bc.jpg)

The Name Space here is Template.Model and in your _ViewImports you have PiranhaTest.Models

Either change the names space in the view imports to match the class or change the standardpage class name space to PiranhaTest.Models

That should at least clear up some of your errors

**Reply by tidyui** [reply]
Yes! If you copy code between projects you have to make sure that you update the namespaces everywhere unless you of course **want** to hang on to the old namespace (see my comments in the gitter chat).

**Reply by dblood2** [reply]
Yep, I had a feeling that's what the issue was, but wanted to see the code first.  You guys didn't need me at all!

**Reply by tidyui** [reply]
@dblood2 We need you more than you know :) Everyone's opinion and input is HIGHLY appreciated! Thanks for just taking the time and moving the discussion here from Gitter!

**Comment by irdublu** [comment]
I feel really bad now because on the other page I had written, I had the namespace right.
I've corrected the name space and those errors have gone and I'm back to the original situation of it not working, before I copied code in desperation.   I can't imagine the image gives you anything new to work on for me.

![404](https://user-images.githubusercontent.com/19719089/107086280-b10b0200-67f1-11eb-9748-5a113d920a68.JPG)

**Reply by tidyui** [reply]
Like I wrote in the Gitter chat. It all depends on the data in your database. If you haven't created any pages this means a `404` is totally valid as the requested url can't be found. If you've copied a database where that saved pages have a different namespace than your current application then the pages won't be found either, in which case a `404` is also totally valid.

**Reply by tidyui** [reply]
The templates we provide are just **templates**, they are not part of the product and you can't just move data or other stuff between projects without having to migrate your data. This is just how the framework works.

Best regards

Håkan

**Comment by irdublu** [comment]
OK, this is beginning to make some sense.  I'm used to tables that contain just data and had not appreciated that namespaces could be involved.  I'll make some data, seed it and see how I get on.
Thank you very much for pointing me in the right direction.  Have to admit I am struggling with this and haven't even see the CMS in action yet, which is of course, the whole reason for it all.
Sorry about the Namespace error earlier.  That was a most unfortunate one-off.  Typical it should happen when I took the image for you.

**Reply by GangsterGeek89** [reply]
Glad you are making progress with it and we all make mistakes sometimes we get to involved in the problem too much and don't see what is right in front of us :)

**Reply by dblood2** [reply]
Now that you have the namespaces tidied up, if you go to the manager, you should see your page types when you attempt to "Add Page".  If I remember correctly, for me it was that simple.  I just added a page titled "Home", edited that - adding a content block, typed "Hello World", Saved, Published, and the 404 was gone.   At least I think that's all I did. 

Let us know how you get on.

**Reply by GangsterGeek89** [reply]
I Ended up getting curious as like I said earlier on gitter ive never bothered to start from an empty project and so I went at it to see what errors if any I encountered. Like @irdublu I was getting the page not found error but was able to login to the manager and see my page type. Once I created the first page the error went away just as @dblood2 mentioned above. Anyway as I was at it I decided to create a repository and create a commit for each step I went through. I have made the repository public so if anyone is having any problems they can browse the commits and look at the limited simple notes I have made and hopefully it may help someone or even me in the future lol. 

Here is a link to the repository : https://github.com/GangsterGeek89/MyPiranhaRazorExample
@tidyui Hope posting a link isn't against any community rules if it is just let me know and I will happily remove it :)

**Reply by irdublu** [reply]
Just had to come back and say thank you again to you all as I've created a page and have seen it in the browser.  Such a small thing you might think but I was going in circles with trying stuff and making it worse.  I'd tried stripping it back to virtually nothing but it was my misunderstanding the role of the database that foiled me.

**Reply by GangsterGeek89** [reply]
Happy to help when I can and glad you got it sorted :)
